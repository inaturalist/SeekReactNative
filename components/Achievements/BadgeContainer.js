// @flow

import * as React from "react";
import { View } from "react-native";

import { viewStyles } from "../../styles/badges/achievements";

type Props = {
  data: Array<Object>,
  renderItem: ( Object ) => any
}

const BadgeContainer = ( { data, renderItem }: Props ): React.Node => (
  <View testID="badge-container" style={[viewStyles.imageContainer]}>
    {data.map( item => (
      <View key={`${item.name}-${item.earnedIconName}`}>
        {renderItem( item )}
      </View>
    ) )}
  </View>
);

export default BadgeContainer;
