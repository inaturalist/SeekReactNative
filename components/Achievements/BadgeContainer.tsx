import * as React from "react";
import { View } from "react-native";

import { viewStyles } from "../../styles/badges/achievements";

interface Badge {
  name: string;
  earnedIconName: string;
}

interface Props {
  data: Badge[];
  renderItem: ( item: Badge ) => any;
}

const BadgeContainer = ( { data, renderItem }: Props ) => (
  <View testID="badge-container" style={[viewStyles.imageContainer]}>
    {data.map( ( item, index ) => (
      <View key={`${item.name}-${item.earnedIconName}-${index}`}>
        {renderItem( item )}
      </View>
    ) )}
  </View>
);

export default BadgeContainer;
