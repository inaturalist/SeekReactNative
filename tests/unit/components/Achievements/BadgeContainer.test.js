import React from "react";
import { render, screen } from "@testing-library/react-native";
import BadgeContainer from "../../../../components/Achievements/BadgeContainer";

describe( "BadgeContainer", () => {
  test( "should render correctly", () => {
    const data = [{ name: "test", earnedIconName: "test" }];
    const mockFn = jest.fn();
    render( <BadgeContainer data={data} renderItem={mockFn} /> );
    expect( screen.findByTestId( "badge-container" ) ).toBeTruthy();
    expect( mockFn ).toHaveBeenCalledTimes( 1 );
    expect( screen ).toMatchSnapshot();
  } );
} );
