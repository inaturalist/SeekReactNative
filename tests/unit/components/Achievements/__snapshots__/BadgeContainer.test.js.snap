// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgeContainer should render correctly 1`] = `
<View
  style={
    [
      {
        "alignSelf": "center",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "width": 377.20734908136484,
      },
    ]
  }
  testID="badge-container"
>
  <View />
</View>
`;
